src/util/utility-types.ts

```diff
@@ -4,3 +4,23 @@
  * Required since React.ComponentProps doesn't think it's a JSXElementConstructor/ReactElement. https://github.com/FB-PLP/traject/pull/11929/files#r948531375
  */
 export type ExtractProps<T> = T extends React.ComponentType<infer P> ? P : T;
+
+/**
+ * Given two types, only allow one of them to be valid
+ */
+export type Only<T, U> = {
+  [P in keyof T]: T[P];
+} & {
+  [P in keyof U]?: never;
+};
+
+/**
+ * Given two types, specify that only one of them can be in
+ * the structure at a given time
+ */
+export type Either<T, U> = Only<T, U> | Only<U, T>;
```

Comment:
Also, how is this different from `Only`?

Both read to me that they're basically `XOR`

Reply:
nvm